tests/test_dvc.py

```diff
@@ -90,14 +92,21 @@ def test_exp_save_on_end(tmp_dir, mocker, save):
     dvc_repo.index.stages = []
     dvc_repo.scm.get_rev.return_value = "current_rev"
     dvc_repo.scm.get_ref.return_value = None
+    (tmp_dir / "data").write_text("data")
     with mocker.patch("dvclive.live.get_dvc_repo", return_value=dvc_repo):
         live = Live(save_dvc_exp=save)
+        live.log_artifact("data")
```

Comment:
Would be nice to add more tests for other scenarios:

- without `save_dvc_exp`
- Calling `log_artifact` twice so we also test the `_dvc_repo.commit` 
- Calling `log_artiact` with `_inside_dvc_exp`.
- Testing or the `OutputDuplicationError` (regardless of whether we raise exception or warning)